### PR TITLE
feat: add validation for wallet start to check first wallet address

### LINKED ISSUE #48: Add first address validation when starting the wallet
Question: should this be required or optional?

Lib PR: https://github.com/HathorNetwork/hathor-wallet-lib/pull/179

### PR DESCRIPTION
**Pull request by pedroferreira1**

### CODE REVIEW

File: src/api/wallet.ts [hunk 1]
```diff
@@ -71,6 +72,8 @@ export const get: APIGatewayProxyHandler = async (event) => {
 const loadBodySchema = Joi.object({
   xpubkey: Joi.string()
     .required(),
+  address0: Joi.string()
```
Comment (msbrogli) [inline src/api/wallet.ts]: I called it `firstAddress` in hathor-wallet-headless. Should we use the same name here?

File: src/api/wallet.ts [hunk 2]
```diff
@@ -111,6 +114,15 @@ export const load: APIGatewayProxyHandler = async (event) => {
     return closeDbAndGetError(mysql, ApiError.WALLET_ALREADY_LOADED, { status });
   }
 
+  const expectedAddress0 = value.address0;
+
+  const address0 = hathorLib.walletUtils.getAddressAtIndex(xpubkey, 0, 0, 'mainnet');
```
Comment (msbrogli) [inline src/api/wallet.ts]: We should get the network from config, right?

Comment (msbrogli) [inline src/api/wallet.ts]: Maybe we should make it optional in the config file. What do you think?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: src/api/wallet.ts (hunk 2 above)
Comment (pedroferreira1) [reply src/api/wallet.ts]: Yes, done

File: src/api/wallet.ts (hunk 1 above)
Comment (pedroferreira1) [reply src/api/wallet.ts]: Makes sense, changed.

File: src/api/wallet.ts (hunk 2 above)
Comment (pedroferreira1) [reply src/api/wallet.ts]: Makes sense, done

**Review by msbrogli — APPROVED** [review]

**Review by msbrogli — APPROVED** [review]

File: src/api/wallet.ts [hunk 3]
```diff
@@ -71,6 +72,7 @@ export const get: APIGatewayProxyHandler = async (event) => {
 const loadBodySchema = Joi.object({
   xpubkey: Joi.string()
     .required(),
+  firstAddress: Joi.string(),
```
Comment (andreabadesso) [inline src/api/wallet.ts]: When `process.env.CONFIRM_FIRST_ADDRESS` is `'true'`, shouldn't `firstAddress` be required?

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/api/wallet.ts (hunk 3 above)
Comment (pedroferreira1) [reply src/api/wallet.ts]: Done. Can you see if it's good?

Comment (andreabadesso) [reply src/api/wallet.ts]: LGTM

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**